msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IMDSRetryPolicy.java

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+//IMDS uses a different try policy than other MI flows, see https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/docs/imds_retry_based_on_errors.md
+class IMDSRetryPolicy extends ManagedIdentityRetryPolicy {
+    private static final int LINEAR_RETRY_NUM = 7;
+    private static final int LINEAR_RETRY_DELAY_MS = 10000; // 10 seconds
+    private static final int EXPONENTIAL_RETRY_NUM = 3;
+    private static final int EXPONENTIAL_RETRY_DELAY_MS = 1000; // 1 second
```

Comment:
I don't think these should be static anymore since they're no longer referenced outside of this file